clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -411,9 +415,36 @@ public void closeVeniceWriters(boolean doFlush) {
   }
 
   @Override
-  protected void closeVeniceViewWriters() {
+  protected void closeVeniceViewWriters(boolean doFlush) {
     if (!viewWriters.isEmpty()) {
-      viewWriters.forEach((k, v) -> v.close());
+      long gracefulCloseDeadline = time.getMilliseconds() + VIEW_WRITER_CLOSE_TIMEOUT_IN_MS;
+      viewWriters.forEach((k, v) -> v.close(doFlush));
+      // Short circuit last VT produce call future if it's incomplete to unblock any consumer thread(s) that are waiting
+      for (PartitionConsumptionState pcs: partitionConsumptionStateMap.values()) {
+        CompletableFuture<Void> lastVTProduceCallFuture = pcs.getLastVTProduceCallFuture();
```

Comment:
I am inclined to mark `lastVTProduceCallFuture` as volatile due to visibility reason. This pointer can be set and get in different threads so it's better to be a `volatile`, otherwise, it's not guaranteed to see its most up-to-date value.